pkg/kubeconfig/auth.go

```diff
@@ -34,8 +34,11 @@ func GetAuthType(config *clientcmdapi.Config, contextName string) AuthType {
 	case authInfo.AuthProvider != nil:
 		return AuthTypeAuthProvider
 	case len(authInfo.ClientCertificate) > 0:
+		return AuthTypeClientCertificate
```

Comment:
Somehow the way this switch was written would return `unknowntype` immediately when a clientCertificate was present. I assume that was not the original intention so I fixed it. :see_no_evil: